### PR TITLE
Introduce a quick way to test across platforms

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: packages/flutter/lib/src/foundation/platform.dart [hunk 1]
```diff
@@ -52,3 +57,21 @@ TargetPlatform get defaultTargetPlatform {
...
 }
+/// Override the [defaultTargetPlatform].
+///
+/// Setting this to null returns the [defaultTargetPlatform] to its original
+/// value (based on the actual current platform).
+///
+/// This setter is only available in debug mode. To change the target platform
+/// in release builds, use the platform override APIs (such as
+/// [ThemeData.platform] in the material library).
+set defaultTargetPlatform(TargetPlatform value) {
```
Comment (abarth) [inline packages/flutter/lib/src/foundation/platform.dart]: I'd make this a real function (e.g., `debugSetDefaultTargetPlatformOverride`) and throw a `NotImplemented` or some other normal sort of exception in release mode.

**Comment by abarth** [comment]
What about the code that checks that all the debug flags are off?  Seems like it should check that this one is clean too.

**Comment by abarth** [comment]
lgtm

**Comment by Hixie** [comment]
Addressed comments. Checking in.

**Pull request merged**